include/bm/bm_sim/field_lists.h

```diff
@@ -24,12 +24,17 @@
 #define BM_BM_SIM_FIELD_LISTS_H_
 
 #include <boost/functional/hash.hpp>
+#include <boost/variant.hpp>
 
 #include <utility>  // for pair<>
 #include <vector>
 #include <unordered_set>
+#include <string>
 
 #include "phv_forward.h"
+#include "bytecontainer.h"
+
+using bm::ByteContainer;
```

Comment:
Not needed as you only manipulate `ByteContainer` in the `bm` namespace to start with. Also `using` declarations are bad practice in headers like this.